Nieuwe RequestLogs (RequestLogRevamp)

safouuwa opened the pull request:
De logs van de requests zijn nu dusdanig aangepast, naar wens van de PO.

Ten eerste volgt het nu de gewenste format van de PO: datetime | request | httpmethod | source_id | data before | data after.
Vervolgens worden de logs nu ook opgeslagen in een .log bestand inplaatsvan een .txt bestand (ook naar wens van de PO)
Tot slot, de logs exclusief gemaakt voor requests die een body hebben (maar nogsteeds alleen voor Transfers, Shipments, Items en Orders, naar wens van de PO)

Om dit allemaal mogelijk te maken, heb ik nu gebruik gemaakt van Filters (voor de middleware die het loggen waar zal maken), Attributes (om specifieke endpoints te markeren die gelogd moeten worden) en Interfaces (die ervoor zorgen dat de controllers in kwestie extra velden bevatten voor data in de logline).

Ook de Filtering endpoints for de RequestLogs zijn nu aangepast.

De grootste wijziging ligt hem hier in de zoekcriteria. Oorspronkelijk kon er gezocht worden naar de user, maar sinds de PO dat liever niet meer in de log wilde hebben, is dat nu verplaatst door het kunnen zoeken van de HTTPmethod. De model en date parameter zijn wel hetzelfde gebleven, sinds die nog wel in de log file terug gekeerd zijn.

Tot slot de bestaande Requestlog Tests aangepast zodat ze van toepassing zullen zijn voor de nieuwe RequestLogs. Zowel Non Happy and Happy path is er nogsteeds in verwerkt, en de asserts voor users zijn gewijzigd naar asserts voor httpmethods. Deze zijn allemaal geslaagd.

safouuwa committed:
.Gitignore update to not include .logs

safouuwa committed:
Initial RequestLog Revamp

De logs van de requests zijn nu dusdanig aangepast, naar wens van de PO.

Ten eerste volgt het nu de gewenste format van de PO: datetime | request | httpmethod | source_id | data before | data after.

Vervolgens worden de logs nu ook opgeslagen in een .log bestand inplaatsvan een .txt bestand (ook naar wens van de PO)

Tot slot, de logs exclusief gemaakt voor requests die een body hebben (maar nogsteeds alleen voor Transfers, Shipments, Items en Orders, naar wens van de PO)

Om dit allemaal mogelijk te maken, heb ik nu gebruik gemaakt van Filters (voor de middleware die het loggen waar zal maken), Attributes (om specifieke endpoints te markeren die gelogd moeten worden) en Interfaces (die ervoor zorgen dat de controllers in kwestie extra velden bevatten voor data in de logline).

safouuwa committed:
RequestLog Filtering Update

Ook de Filtering endpoints for de RequestLogs zijn nu aangepast.

De grootste wijziging ligt hem hier in de zoekcriteria. Oorspronkelijk kon er gezocht worden naar de user, maar sinds de PO dat liever niet meer in de log wilde hebben, is dat nu verplaatst door het kunnen zoeken van de HTTPmethod. De model en date parameter zijn wel hetzelfde gebleven, sinds die nog wel in de log file terug gekeerd zijn.

safouuwa committed:
Swagger Fix

Bepaalde methods verwijderd die het laden van het Swagger documentatie niet mogelijk maakte. De methods in kwestie zijn nu verplaatst door properties.

safouuwa committed:
Nieuwe RequestLog Tests

Tot slot de bestaande Requestlog Tests aangepast zodat ze van toepassing zullen zijn voor de nieuwe RequestLogs. Zowel Non Happy and Happy path is er nogsteeds in verwerkt, en de asserts voor users zijn gewijzigd naar asserts voor httpmethods. Deze zijn allemaal geslaagd.

linkverk commented:
Bij checken van deze PR ik heb nagekeken werking van de geupdated RequestLog en werking van deze code.. Het was echt goed gedaan. Volgens format van PO, alles goed gedaan. Hieronder staan een van test voor request en die updated in realtime. 
![Schermafbeelding 2025-01-13 172829](https://github.com/user-attachments/assets/e4c0eee8-50d7-4049-a03d-cda8730c8872)
![Schermafbeelding 2025-01-13 172909](https://github.com/user-attachments/assets/9c7c1c16-dbfd-449e-b6e0-60e64d82435e)
Alleen ik heb wel gezien dat als ik dit doe request in de log file komt niet relevante tijd. Dus ik had om 17:23 gedaan. Maar in de log file staat 16:23.

safouuwa committed:
Datetime aangepast

1 uur gevoegd aan utc, om onze eigen Nederlandse tijd te krijgen in de logfile

safouuwa commented:
Tijd aangepast: 1 uur gevoegd aan utc, om onze eigen Nederlandse tijd te krijgen in de logfile.

safouuwa committed:
New Source_ID Update

In plaats van IP address in de logline te verwerken, wordt er nu gewerkt met een Source_ID die van de headers uitgelezen wordt (naar wens van PO).

safouuwa commented:
In plaats van IP address in de logline te verwerken, wordt er nu gewerkt met een Source_ID die van de headers uitgelezen wordt (naar wens van PO).

safouuwa committed:
Emptied out Requestlog

linkverk commented:
Ik zie tijd werd gefixt en dat er werd in plaats van IP, een source_id gezet volgens wens van PO. Goed gedaan
![Schermafbeelding 2025-01-14 112423](https://github.com/user-attachments/assets/af8f55c5-579a-4044-b42b-afe1eb0f08a6)
![Schermafbeelding 2025-01-14 112342](https://github.com/user-attachments/assets/48043f51-a0bf-42a9-8357-2e64aff26698)
